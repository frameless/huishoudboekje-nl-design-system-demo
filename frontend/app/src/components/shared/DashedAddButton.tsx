import {AddIcon} from "@chakra-ui/icons";
import {Button, ButtonProps} from "@chakra-ui/react";
import React from "react";
import {useTranslation} from "react-i18next";

type DashedAddButtonProps = Pick<ButtonProps, "type" | "isLoading" | "isDisabled"> & {
	onClick?: VoidFunction,
	children?: JSX.Element | string
};

const DashedAddButton: React.FC<DashedAddButtonProps> = ({onClick, children}) => {
	const {t} = useTranslation();

	return (
		<Button colorScheme={"primary"} borderStyle={"dashed"} variant={"outline"} leftIcon={<AddIcon />} w={"100%"} h={"100%"} borderRadius={5} p={5} onClick={onClick}>
			{children || t("global.actions.add")}
		</Button>
	);
};

export default DashedAddButton;