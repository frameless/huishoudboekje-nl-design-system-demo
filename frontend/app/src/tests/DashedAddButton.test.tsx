import {fireEvent, getByText, render} from "@testing-library/react";
import React from "react";
import DashedAddButton from "../components/shared/DashedAddButton";

jest.mock("react-i18next", () => require("./utils/mock-hooks").reactI18NextMock());
jest.mock('@chakra-ui/react', () => ({
	...jest.requireActual('@chakra-ui/react'),
	Button: ({ children, onClick, ...rest }) => (
	  <button onClick={onClick} {...rest}>
		{children}
	  </button>
	),
  }));

describe("Dashed button", () => {
	it("Show dashed button with label", () => {
		const label = "Toevoegen";
		const {container} = render(<DashedAddButton>{label}</DashedAddButton>);

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.textContent).toBe(label);
	});

	it("Show dashed button with default label", () => {
		const {container} = render(<DashedAddButton />);

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.textContent).toBe("global.actions.add");
	});

	it("Check the onClick", () => {
		const onClick = jest.fn();
		const label = "Opslaan";

		const {container} = render(<DashedAddButton onClick={onClick}>{label}</DashedAddButton>);

		expect(container.innerHTML).toMatchSnapshot();
		expect(container.textContent).toBe("Opslaan");

		const clickEvent = new Event("click", {
			bubbles: true,
			cancelable: true,
		});

		const element = getByText(container, "Opslaan");

		fireEvent(element, clickEvent);
		expect(onClick).toHaveBeenCalledTimes(1);

		fireEvent(element, clickEvent);
		expect(onClick).toHaveBeenCalledTimes(2);

	});
});