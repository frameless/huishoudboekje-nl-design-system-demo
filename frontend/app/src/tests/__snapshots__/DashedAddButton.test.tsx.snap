// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashed button Check the onClick 1`] = `"<button type=\\"button\\" class=\\"chakra-button css-1c6t2lx\\"><span class=\\"chakra-button__icon css-1wh2kri\\"><svg viewBox=\\"0 0 24 24\\" focusable=\\"false\\" class=\\"chakra-icon css-onkibi\\" aria-hidden=\\"true\\"><path fill=\\"currentColor\\" d=\\"M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z\\"></path></svg></span>Opslaan</button>"`;

exports[`Dashed button Show dashed button with default label 1`] = `"<button type=\\"button\\" class=\\"chakra-button css-1c6t2lx\\"><span class=\\"chakra-button__icon css-1wh2kri\\"><svg viewBox=\\"0 0 24 24\\" focusable=\\"false\\" class=\\"chakra-icon css-onkibi\\" aria-hidden=\\"true\\"><path fill=\\"currentColor\\" d=\\"M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z\\"></path></svg></span>global.actions.add</button>"`;

exports[`Dashed button Show dashed button with label 1`] = `"<button type=\\"button\\" class=\\"chakra-button css-1c6t2lx\\"><span class=\\"chakra-button__icon css-1wh2kri\\"><svg viewBox=\\"0 0 24 24\\" focusable=\\"false\\" class=\\"chakra-icon css-onkibi\\" aria-hidden=\\"true\\"><path fill=\\"currentColor\\" d=\\"M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z\\"></path></svg></span>Toevoegen</button>"`;
